DEVELOPER.md

```diff
@@ -0,0 +1,359 @@
+# Developer Guide for Testsys
+
+## What is testsys
+
+Testsys is a framework that leverages kubernetes (k8s) to create resources and run tests.
+To learn more about how it works, see the [design](design/DESIGN.md) document.
+
+### Navigating the `bottlerocket-test-system` repo
+
+#### [Makefile.toml](Makefile.toml)
+
+Testsys uses make to unify the logic needed to build testsys images.
+Any image can be built using `make <desired image>`.
+To make all bottlerocket related images, `make images` can be used, then `make tag-images` can be used to give all images the same tag and `make publish-images` can be used to publish all images.
+
+#### [Dockerfile](Dockerfile)
+
+Testsys relies on docker images to run tests and create resources.
+The Dockerfile contains the logic needed to create all bottlerocket related images.
+
+#### [Agent](agent)
+
+Testsys is broken up into 3 main things: the controller, test agent, and resource agent.
+
+[agent-common](agent/agent-common) contains logic that is related to both the test and resource agents.
+
+[test-agent](agent/test-agent) provides the framework for creating test agents.
+To create a test agent, the [`Runner` trait](agent/test-agent/src/lib.rs#L51) needs to be implemented.
+Check out the [example test agent](agent/test-agent/examples/example_test_agent/main.rs) to see how to create a test agent.
+
+[resource-agent](agent/resource-agent) provides the framework for creating resource agents.
+To create a resource agent, the [`Create` trait](agent/resource-agent/src/mod.rs#L37) and [`Destroy` trait](agent/resource-agent/src/mod.rs#L73) need to be implemented.
+Check out the [example resource agent](agent/resource-agent/examples/example_resource_agent/main.rs) to see how to create a resource agent.
+
+### [Bottlerocket](bottlerocket)
+
+The bottlerocket directory of the repo is broken into 3 sections: agents, testsys, and types.
+
+[agents](bottlerocket/agents) contains the implementations of all bottlerocket agents.
+
+The types used by the agents can be found in [types](bottlerocket/types).
+
+[testsys](bottlerocket/testsys) contains a cli implementation that is closely related to bottlerocket agents but will be removed in the near future.
+
+### [Cli](cli)
+
+Cli is and example command line interface that uses the [TestManager](model/src/test_manager/manager.rs) to interact with a testsys cluster.
+
+### [Controller](controller)
+
+The controller orchestrates the the interaction between test and resource. For more info see the [design](design/DESIGN.md) document. 
```

Comment:
Double word:

```diff
- The controller orchestrates the the interaction between test and resource. For more info see the [design](design/DESIGN.md) document.
+ The controller orchestrates the interaction between test and resource. For more info see the [design](design/DESIGN.md) document.
```

This is a bit confusing for me as well. What is a test and a resource in this context?